### PR TITLE
Add a license check script

### PR DESCRIPTION
**Pull request by pento**
To ensure that any modules we use are GPL2 compatible, we need a script to scan across them and check their licenses.

### CODE REVIEW

File: packages/scripts/scripts/check-licenses.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		return;
+	}
+
+	const filename = path + '/package.json';
+	if ( ! existsSync( filename ) ) {
+		console.error( `Unable to locate package.json in ${path}.` );
+		process.exit( 1 );
+	}
+
+	const package = require( filename );
```
Comment (gziolo) [inline packages/scripts/scripts/check-licenses.js]: `package` is a reserver work, `pkg` isn't as cool, but common in node world.

File: packages/scripts/scripts/check-licenses.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
+// Do this as the first thing so that any code reading it knows the right env.
+process.env.BABEL_ENV = 'development';
```
Comment (gziolo) [inline packages/scripts/scripts/check-licenses.js]: Those 10 lines at the beginning are Jest specific - they can be removed probably.

File: packages/scripts/scripts/check-licenses.js (hunk 1 above)
Comment (ntwb) [reply packages/scripts/scripts/check-licenses.js]: Did. No. Know. That.

File: packages/scripts/scripts/check-licenses.js (hunk 2 above)
Comment (pento) [reply packages/scripts/scripts/check-licenses.js]: Ah, cool. I assumed they were necessary, but I'll get rid of them.

**Comment by ntwb** [comment]
Issue moved to [WordPress/gutenberg #7822](https://github.com/WordPress/gutenberg/issues/7822) via [**ZenHub**](https://www.zenhub.com/)

**Pull request closed without merging**